azurerm/resource_arm_private_link_service_test.go

```diff
@@ -0,0 +1,309 @@
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.private_ip_allocation_method", "Static"),
+					resource.TestCheckResourceAttr(resourceName, "load_balancer_frontend_ip_configurations.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.env", "test"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPrivateLinkService_update(t *testing.T) {
+	resourceName := "azurerm_private_link_service.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateLinkServiceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateLinkService_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateLinkServiceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "fqdns.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "fqdns.0", "testFqdns"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.subnet.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.private_ip_address", "10.5.1.17"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.private_ip_address_version", "IPv4"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.private_ip_allocation_method", "Static"),
+					resource.TestCheckResourceAttr(resourceName, "load_balancer_frontend_ip_configurations.#", "1"),
+				),
+			},
+			{
+				Config: testAccAzureRMPrivateLinkService_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateLinkServiceExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "fqdns.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "fqdns.0", "testFqdns2"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.subnet.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.private_ip_address", "10.5.1.17"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.private_ip_address_version", "IPv4"),
+					resource.TestCheckResourceAttr(resourceName, "ip_configurations.0.private_ip_allocation_method", "Static"),
+					resource.TestCheckResourceAttr(resourceName, "load_balancer_frontend_ip_configurations.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.env", "test"),
+				),
+			},
```

Comment:
Can we add a step the updates it back to the basic configuration and add a import verification step too?

Reply:
Updated code.